packages/babel-core/test/fixtures/transformation/es6.parameters/rest-member-expression-optimisation/expected.js

```diff
@@ -1,9 +1,9 @@
 var t = function () {
-    var x = arguments[0];
-    var y = arguments[1];
+    var x = arguments.length <= 0 || arguments[0] === undefined ? undefined : arguments[0];
```

Comment:
You're right, it's not. In most cases `arguments.length <= 0 || (undefined || arguments[0])` (I suppose that's what you suggest?) behaves the same way, but they are not really equivalent. Consider the following:

```javascript
var _arguments = {length: 0};
console.log(_arguments.length <= 0 || _arguments[0] === undefined ? undefined : _arguments[0]);
console.log(_arguments.length <= 0 || (undefined || _arguments[0]));
```
First case will return `undefined`, second case will return `true`.